tm2/pkg/p2p/dial/dial.go

```diff
@@ -0,0 +1,83 @@
+package dial
+
+import (
+	"sync"
+	"time"
+
+	"github.com/gnolang/gno/tm2/pkg/p2p/types"
+	queue "github.com/sig-0/insertion-queue"
+)
+
+// Item is a single dial queue item, wrapping
+// the approximately appropriate dial time, and the
+// peer dial address
+type Item struct {
+	Time    time.Time         // appropriate dial time
+	Address *types.NetAddress // the dial address of the peer
+}
+
+// Less is the comparison method for the dial queue Item (time ascending)
+func (i Item) Less(item Item) bool {
+	return i.Time.Before(item.Time)
+}
+
+// Queue is a time-sorted (ascending) dial queue
+type Queue struct {
```

Comment:
Is that Queue structure really needed? It feels too complex for a slice/set of NetAddresses. I understand you are ordering by dial time, but why is that needed?